subsys/bluetooth/mesh/lpn.c

```diff
@@ -222,8 +222,12 @@ static void clear_friendship(bool force, bool disable)
 		lpn->old_friend = lpn->frnd;
 	}
 
-	if (lpn_cb && lpn->frnd != BT_MESH_ADDR_UNASSIGNED) {
-		lpn_cb(lpn->frnd, false);
+	if (lpn->frnd != BT_MESH_ADDR_UNASSIGNED) {
+		bt_mesh_heartbeat_send();
```

Comment:
If I'm reading this correctly, I think there's a problem here. As `lpn_set_state()` isn't called yet, we'll be populating the heartbeat with the old LPN state here, mistakenly sending out a heartbeat with the pre-clear LPN flag, instead of the new one.